src/runtime/TemporalObject.cpp

```diff
@@ -1245,6 +1335,33 @@ int TemporalPlainDateObject::compareISODate(int firstYear, int firstMonth, int f
     return 0;
 }
 
+std::map<TemporalObject::DateTimeUnits, int> TemporalPlainDateObject::regulateISODate(ExecutionState& state, int year, int month, int day, const Value& overflow)
+{
+    if (overflow.asString()->equals(state.context()->staticStrings().lazyConstrain().string())) {
+        if (month > 12) {
+            month = 12;
+        } else if (month < 1) {
+            month = 1;
+        }
+
+        int daysInMonth = TemporalCalendarObject::ISODaysInMonth(state, year, month).asInt32();
```

Comment:
This should be checked if day > 28 for faster performance. Every month has at least 28 days.